src/variables.rs

```diff
@@ -515,3 +606,16 @@ pub async fn set_pipeline_vars_from_file(
         process::exit(2);
     }
 }
+
+fn find_duplicates<T: Eq + Hash + Clone>(vec: Vec<T>) -> Vec<T> {
+    let mut seen = HashSet::new();
+    let mut duplicates = HashSet::new();
+
+    for item in vec {
+        if !seen.insert(item.clone()) {
+            duplicates.insert(item);
+        }
+    }
+
+    duplicates.into_iter().collect()
```

Comment:
here we can use a vector directly and do not need to convert the duplicates into an iter and collect it then.